### DISCUSSION TITLE
When shoud I install deps in root and when in certain package?

### DISCUSSION THREAD

**Discussion by niksimFS24** [vercel/turborepo#5493] [Help] [answered]
Hello everyone :) I could not find this info anywhere so I am here for help.

Let's say I have couple of apps and also some configs as well. What are the rules when should I install any of the deps in the root and when in certain package only?

Since I would like to have installed packages versions the same across all the apps and configs, would it make sense to install everything in the root actually? That sounds most reasonable to me.

I do not understand why would I need `eslint` packages installed in the root and also in the packages as well etc? Where should I install packages related to my `jest` configs if I want to have it as a separate package?

Any insight is welcome, thanks a lot! :)

**Comment by masterbater** [comment, marked as answer]
Most of the time you only need to release one of the packages or one of your apps in production hence the dependencies in each individiual apps, or else if you install in root, then you need to build all

**Reply by anthonyshew** [reply]
We have also created new documentation around this for best practices: https://turbo.build/repo/docs/crafting-your-repository/managing-dependencies
